### PR TITLE
store/tikv: tiny refactoring, change `twoPhaseCommitAction` to interface

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

We define ` type twoPhaseCommitAction int` and use `switch case` on them.

```
func (ca twoPhaseCommitAction) String() string {
	switch ca {
	case actionPrewrite:
		return "prewrite"
	case actionCommit:
		return "commit"
	case actionCleanup:
		return "cleanup"
	case actionPessimisticLock:
		return "pessimistic_lock"
	case actionPessimisticRollback:
		return "pessimistic_rollback"
	}
	return "unknown"
}
```

```
func (c *twoPhaseCommitter) getProcFuncByType(action twoPhaseCommitAction) (procOneBatchFn, error) {
	var singleBatchActionFunc procOneBatchFn
	switch action {
	case actionPrewrite:
		singleBatchActionFunc = c.prewriteSingleBatch
	case actionCommit:
		singleBatchActionFunc = c.commitSingleBatch
	case actionCleanup:
		singleBatchActionFunc = c.cleanupSingleBatch
	case actionPessimisticLock:
		singleBatchActionFunc = c.pessimisticLockSingleBatch
	case actionPessimisticRollback:
		singleBatchActionFunc = c.pessimisticRollbackSingleBatch
	default:
		return nil, errors.Errorf("invalid action type=%v", action)
	}
	return singleBatchActionFunc, nil
}
```

I need to closure some data on `pessimisticLockSingleBatch`, but I think it's improper to change the 

```
type procOneBatchFn func(bo *Backoffer, batch batchKeys) error
```

to something like 

```
func(bo *Backoffer, killed *uint32, batch batchKeys) 
```

so I think using interface is more flexible.

### What is changed and how it works?

change `twoPhaseCommitAction` to interface. 

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch
This refactoring is prepared for an incoming bug-fix PR

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @coocood @lysu @cfzjywxk

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -38,14 +38,23 @@ import (
...
-	actionPrewrite twoPhaseCommitAction = 1 + iota
-	actionCommit
-	actionCleanup
-	actionPessimisticLock
-	actionPessimisticRollback
+type actionPrewrite struct{}
+type actionCommit struct{}
+type actionCleanup struct{}
+type actionPessimisticLock struct{}
+type actionPessimisticRollback struct{}
```
Comment (lysu) [inline store/tikv/2pc.go]:
```
type (
   actionPrewrite struct{}
   actionCommit struct{}
   actionCleanup struct{}
   actionPessimisticLock struct{}
   actionPessimisticRollback struct{}
)
```
we can avoid duplicate `type` words~

**Comment by cfzjywxk** [comment]
LGTM

**Comment by crazycs520** [comment]
@tiancaiamao  Need cherry pick to 3.1?

**Pull request merged**